pkg/roachprod/logger/log_redirect_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package logger
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/stretchr/testify/require"
+)
+
+type mockLogger struct {
+	logger *Logger
+	writer *mockWriter
+}
+
+type mockWriter struct {
+	lines []string
+}
+
+func (w *mockWriter) Write(p []byte) (n int, err error) {
+	w.lines = append(w.lines, string(p))
+	return len(p), nil
+}
+
+func newMockLogger(t *testing.T) *mockLogger {
+	writer := &mockWriter{}
+	logConf := Config{Stdout: writer, Stderr: writer}
+	l, err := logConf.NewLogger("" /* path */)
+	require.NoError(t, err)
+	return &mockLogger{logger: l, writer: writer}
+}
+
+func requireLine(t *testing.T, l *mockLogger, line string) {
+	t.Helper()
+	found := false
+	for _, logLine := range l.writer.lines {
+		if strings.Contains(logLine, line) {
+			found = true
+			break
+		}
+	}
+	require.True(t, found, "expected line not found: %s", line)
+}
+
+func TestLogRedirect(t *testing.T) {
```

Comment:
If the goal of the PR is to make sure we stop leaking roachtest args, then should we have a test for that? Not entirely sure how to easily do that though, simplest thing I can think of is maybe assert that we only see one line in the writer?

Reply:
It's the main goal. Initially I had a goroutine router (scoping log calls for CRDB log to goroutines so it ends up with the test log) as well to maybe make the CRDB log more functional for roachtests. But it ended up feeling like an overly complex solution, so now it's just a barebones interceptor to avoid logging the args. We could technically make args optional in the logger itself, but I'd rather not mess with prod code for this.